src/api/app/controllers/webui/user_controller.rb

```diff
@@ -193,28 +193,22 @@ def change_password
       return
     end
 
-    # check the valid of the params
-    unless User.current.password_equals?(params[:password])
-      errmsg = 'The value of current password does not match your current password. Please enter the password and try again.'
-    end
-    if params[:new_password] != params[:repeat_password]
-      errmsg = 'The passwords do not match, please try again.'
-    end
-    if params[:password] == params[:new_password]
-      errmsg = 'The new password is the same as your current password. Please enter a new password.'
-    end
-    if errmsg
-      flash[:error] = errmsg
-      redirect_to action: :show, user: User.current
-      return
-    end
-
     user = User.current
-    user.update_password params[:new_password]
-    user.save!
 
-    flash[:success] = 'Your password has been changed successfully.'
-    redirect_to action: :show, user: User.current
+    if user.authenticate(params[:password])
```

Comment:
This doesn't work if you still have the old password set!

Reply:
Hm... You're right.

Reply:
Method is overwritten so it's now working with that code. Please double check and remove your and my comment if you agree. It makes the PR more readable then.